Cabal/Distribution/Simple/Program/Builtin.hs

```diff
@@ -309,6 +311,16 @@ hscolourProgram = (simpleProgram "hscolour") {
         _         -> ""
   }
 
+doctestProgram :: Program
+doctestProgram = (simpleProgram "doctest") {
```

Comment:
I think teaching cabal-install how to compile correct versions of haddock and doctest is worthwhile, and fits into the new-build model (we need a simulated build-tools-depends) but it is definitely out of scope for this ticket. Maybe we should make a ticket for it.